### ISSUE TITLE
HtmlText module reset content when switches between Rich text and Html source

### ISSUE THREAD

**Issue by chlupac** [oqtane/oqtane.framework#253]
When I switch between editors, changes in text are lost.

**Comment by chlupac** [comment]
Discovered point where to fill Quill content correctly when switching between html/rich : 
sample from MS

```c#
@using Microsoft.JSInterop
@inject IJSRuntime JSRuntime

<div @ref="divElement">Text during render</div>

@code {
    private ElementReference divElement;

    protected override async Task OnAfterRenderAsync(bool firstRender)
    {
        if (firstRender)
        {
            await JSRuntime.InvokeVoidAsync(
                "setElementText", divElement, "Text after render");
        }
    }
}
```

**Comment by ADefWebserver** [comment]
That pattern is essentially what is being followed in the current code. I don't believe that will solve the underlying issue.

@sbwalker - When I look at what I did in [Blazor Blogs](https://github.com/ADefWebserver/Blazor-Blogs/blob/master/BlazorBlogs/Pages/BlogAdministration.razor), I see that when switching between the editor modes, I coded it like this:

```
    private void RichTextEditor()
    {
        RichTextEditorMode = true;
        StateHasChanged();
    }

    private async Task RawHTMLEditor()
    {
        RichTextEditorMode = false;
        SelectedBlog.BlogContent =
            await this.QuillHtml.GetHTML();
    }
```

**Comment by chlupac** [comment]
I'm waiting for naming update and i will made this fix again.
Here is two problems. 
1. context variable which holds text is not preserved between switch
2. when i fixed it Quill was destroyed after switch, because i'm trying to fill context before it was rendered.

**Comment by chlupac** [comment]
Fixed with #262

**Comment by sbwalker** [comment]
There is a bug in the PR which can be reproduced if you add a new Html/Text module to a page ( which initially has no content ) and you click Edit. It displays the both rich text editor and raw text editor side by side.

Note that the ability to switch between a rich text editor and raw text editor should be self contained in the RichTextditor.razor component - then any module can take advantage of it simply by including the component.

**Comment by chlupac** [comment]
solved in #268

**Comment by sbwalker** [comment]
Thank you for the fix. I will open a new issue related to moving the raw editor logic into the rich text editor component.
